### PR TITLE
extension config management

### PR DESCRIPTION
**Pull request by jonathan-gramain**
ft: manage extensions configuration
    
Each extension is now responsible to manage its config stored in the
global config.json, through a custom configValidator function.

Built on top of https://github.com/scality/backbeat/pull/98

### CODE REVIEW

File: conf/Config.js [hunk 1]
```diff
@@ -31,33 +31,27 @@ class Config {
         const parsedConfig = joi.attempt(config, backbeatConfigJoi,
                                          'invalid backbeat config');
 
+        if (parsedConfig.extensions) {
+            Object.keys(parsedConfig.extensions).forEach(extName => {
+                const index = require(`../extensions/${extName}/index.js`);
```
Comment (rahulreddy) [inline conf/Config.js]: I recommend looping through `extensions` folder and require all extensions at the top.

Comment (jonathan-gramain) [reply conf/Config.js]: If just a style preference, I would rather leave it here as I like the fact that I can just browse a config validated by joi first, guaranteed to have the fields I need to browse.

If there are other good reasons to move `require` to the top (caching or other) I can do the necessary manual checks on the extensions array, then require extensions in a global array and then browse that array here to check each extension config.

**Comment by jonathan-gramain** [comment]
@ironman-machine r+

**Comment by jonathan-gramain** [comment]
@ironman-machine r- r+

**Comment by jonathan-gramain** [comment]
@ironman-machine r- r+

**Comment by jonathan-gramain** [comment]
@ironman-machine r- r+

**Pull request merged**